apps/prairielearn/src/migrations/20231227114715__group_user_roles__foreign_key_group_users.sql

```diff
@@ -0,0 +1,14 @@
+DELETE FROM group_user_roles gur
+WHERE
+  NOT EXISTS (
+    SELECT
+      1
+    FROM
+      group_users gu
+    WHERE
+      gu.group_id = gur.group_id
+      AND gu.user_id = gur.user_id
+  );
+
+ALTER TABLE group_user_roles
+ADD FOREIGN KEY (group_id, user_id) REFERENCES group_users (group_id, user_id) ON DELETE CASCADE;
```

Comment:
We pretty much always want to be explicit about this.

```suggestion
ADD FOREIGN KEY (group_id, user_id) REFERENCES group_users (group_id, user_id) ON DELETE CASCADE ON UPDATE CASCADE;
```

Reply:
Done in 59c2b9b.